tests/testsuite/docscrape.rs

```diff
@@ -48,6 +48,127 @@ fn basic() {
     assert!(p.build_dir().join("doc/src/ex/ex.rs.html").exists());
 }
 
+// This test ensures that even if there is no `[workspace]` in the top-level `Cargo.toml` file, the
+// dependencies will get their examples scraped and that they appear in the generated documentation.
+#[cargo_test(nightly, reason = "rustdoc scrape examples flags are unstable")]
+fn without_workspace() {
```

Comment:
nit: can we make the name more descriptive, like `scrape_examples_for_non_workspace_reexports`?